app/src/editor/canvas/state/linking.js

```diff
@@ -0,0 +1,22 @@
+const LINK_PREFIX = 'Link'
+function validateId(id) {
+    if ((typeof id !== 'string' && typeof id !== 'number') || id === '') {
+        throw new TypeError(`id must be non-empty string|number. Got ${typeof id}: ${id}`)
+    }
+}
+
+export function getLink(parentId) {
+    validateId(parentId)
+    return window.localStorage.getItem(`${LINK_PREFIX}:${parentId}`)
```

Comment:
You could make use of the `isLocalStorageAvailable` helper from `$shared/utils/storage` here.

Reply:
hrm, when is localStorage not available though?

Reply:
Basically the cases here where that helper is derived: https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API/Using_the_Web_Storage_API#Feature-detecting_localStorage

> One example of that is Safari, which in Private Browsing mode gives us an empty localStorage object with a quota of zero, effectively making it unusable.

Maybe it's a non-issue in a real world scenario but might be worth testing that.

Reply:
I'll add the check